skimage/feature/texture.py

```diff
@@ -97,11 +99,21 @@ def greycomatrix(image, distances, angles, levels=256, symmetric=False,
     assert_nD(distances, 1, 'distances')
     assert_nD(angles, 1, 'angles')
 
-    assert levels <= 256
     image = np.ascontiguousarray(image)
     assert image.min() >= 0
-    assert image.max() < levels
-    image = image.astype(np.uint8)
+
+    image_max = image.max()
```

Comment:
From my point of view, the output has to be either full 2**{8/16} x 2**{8/16} matrix of coefficients, or sparse matrix/dictionary with non-zero coefficients only.
We can implement both alternatives and add a parameter for user to choose between them. The default alternative has to be the same as the current one. Otherwise, it will be an API change and we have to follow deprecation path.

> If the maximum value << levels, there will many zeros.
Also, the might be many zeros within the [0:levels] range. As for me, clipping from the upper bound of a range doesn't seem to be the _right_ simplification.

By the way, for uint16, if have set all things correctly, there will be `65536**2 * 2 bytes` matrix, what equals to ~8GB. Not suitable.